sympy/simplify/cse_main.py

```diff
@@ -536,3 +536,101 @@ def cse(exprs, symbols=None, optimizations=None, postprocess=None,
         return replacements, reduced_exprs
 
     return postprocess(replacements, reduced_exprs)
+
+
+class Marker:
+    pass
+
+# XXX: Should this go in sympy.utilities.iterables?
+def shortest_repeated_subsequence(S, ignore=(Marker,)):
+    """
+    Given a sequence S, find the shortest repeated subsequence of length >= 2
+
+    A subsequence is only considered a repeated subsequence if it can't be
+    extended in every instance. For example, [a, b] is not a subsequence of
+    [a, b, c, a, b, c] because every occurrence of [a, b] can be extended to
+    [a, b, c].
+
+    If no repeated subsequence is found, returns None.
+
+    The input sequence should support iteration, indexing, and slicing.
+
+    To use concurrently with multiple sequences, concatenate them separated by
+    markers (custom markers can be pass in through the 'ignore' parameter').
+
+    Examples
+    ========
+
+    In the examples here, we use strings. A more common example for SymPy
+    would be the args of a noncommutative Mul.
+
+    >>> from sympy.simplify.cse_main import shortest_repeated_subsequence
+    >>> s = 'abcabc'
+    >>> shortest_repeated_subsequence(s)
+    'abc'
+    >>> print(shortest_repeated_subsequence('abc'))
+    None
+
+    >>> from itertools import chain
+    >>> from sympy.utilities.iterables import flatten
+    >>> s1, s2, s3 = 'abc', 'babc', 'abca'
+    >>> marker = '$'
+    >>> S = ''.join(chain(*flatten(zip([s1, s2, s3], '$$$'))))[:-1]
```

Comment:
I can't try this now, but what happens if one string has no common substring with the others?